enhancements/microshift/microshift-default-csi-plugin.md

```diff
@@ -0,0 +1,371 @@
+lvmd process reads this once at startup.  Changes to the configuration require that a) the data be pushed to the `lvmd`
+configMap and b) the lvmd process be restarted so that it will pick up the latest configuration.
+
+1. On startup, MicroShift will check for the existence of a file called `/etc/microshift/lvmd.yaml`.  
+   1. If the file exists, MicroShift will read the file into memory
+   1. If the file doesn't exist, MicroShift will assume hardcoded default values.
+1. MicroShift will calculate a SHA256 checksum from the config data. 
+1. MicroShift will generate a ConfigMap to encapsulate the lvmd config data.
+
+    _Example ConfigMap:_
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: lvmd
+      namespace: openshift-storage
+    data:
+      lvmd.yaml: |
+        device-classes:
+        - name: user-provided-config
+          volume-group: rhel
+          default: true
+        socket-name: /run/lvmd/lvmd.sock
+    ```
+
+1. MicroShift will Create / Update the ConfigMap.
+1. MicroShift will render the topolvm-node daemonset template, with the checksum value and `socket-name` specified in
+the pod template. 
+annotation.  This has the following effects:
+   1. If the daemonSet does not exist, it will be started
+   1. Else, if the file changed, the pod will be restarted.
+   1. Else, if the file did not change, the pod will not be restarted
+
+### Risks and Mitigations
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+[//]: # (**Note:** *Section not required until targeted at a release.*)
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+MicroShift is deployed as an application on top of an operating system, and so its upgrade/downgrade model differs
+significantly from OCP's.  Because MicroShift's cluster architecture is embedded within the MicroShift binary, 
+the entire application is upgraded as a whole, distributed by Red Hat as a set of RPMs and container images.  The ART team will generate new
+MicroShift RPMs and make these available to users. Users deploying MicroShift on top of RHEL for Edge are provided a
+recipe for generating and deploying a new rpm-ostree.
+
+Downgrades are handled similarly, where users install an older version of MicroShift, or deploy a rpm-ostree layer
+containing the older version.
+
+Customers deploying onto RHEL for Edge will be able to rely on [Greenboot](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/composing_installing_and_managing_rhel_for_edge_images/index?extIdCarryOver=true&sc_cid=701f2000001OH7EAAW#how-are-rhel-for-edge-images-restored_managing-rhel-for-edge-images) to ensure that if an upgrade/rollback produces
+and unstable operating environment, the system will automatically roll the change back to the last known stable state.
```

Comment:
Another risk: Rollback will not modify the content of any volumes managed by the CSI driver.

Reply:
For posterity:
we'll document this in microshift documentation, as it's an expected behavior in OCP, but unacquainted users may not realize that a system roll back will not rollback workload data on disk.